pose_broadcaster/src/pose_broadcaster.cpp

```diff
@@ -148,40 +160,53 @@ controller_interface::return_type PoseBroadcaster::update(
     realtime_publisher_->unlockAndPublish();
   }
 
-  if (realtime_tf_publisher_ && realtime_tf_publisher_->trylock())
+  if (realtime_tf_publisher_)
   {
-    bool do_publish = false;
-    // rlcpp::Time comparisons throw if clock types are not the same
-    if (tf_last_publish_time_.get_clock_type() != time.get_clock_type())
+    if (!is_pose_valid(pose))
```

Comment:
May be this check can be done outside the realtime_tf_publisher condition?. That way we don't disturb the previous code